### PR TITLE
Function bundle maps

### PR DESCRIPTION
**Pull request by atait**
## Mapping MeasuredFunction methods over FunctionBundles
If an attribute is not found in the FunctionBundle, and it is in it's member, and it is callable, then it will be mapped to every function in the bundle, returning a new bundle.

For example, starting with
```python
spct1 = Spectrum([0, 1, 2], [2, 5, 3])
spct2 = Spectrum([0, 1, 2], [4, 4, 1])
funBun = FunctionBundle([spct1, spct2])
```
You can do
```python
croppedFunBund = FunctionBundle([
                                 spct1.crop([0, 1]),
                                 spct2.crop([0, 1]))
                                ])
funBun.crop([0, 1]) == croppedFunBund  # True
```

### CODE REVIEW

**Comment by thomaslima** [comment]
I am not quite sure about the usefulness of functionbundle, but maybe we can take numpy arrays as inspiration. for example, a spectrum can be represented in a vector of dimension data[2, N], and we access it like this:
x = data[0]
y = data[1]
if we prepare an array of such data, we can have arr[M,2,N]:
data1 = arr[1]
data1x = arr[1,0,:]
to apply a numpy function over any axis, we can do:
arr[1,0,:].crop(0,1)

so in a way we can use `slices` for indexing in higher dimensional data structures, instead of creating these Bundles.

**Comment by atait** [comment]
FunctionBundle is different from a numpy array of them because it makes sure all of the bundled functions have the same x-basis, even if they start with different domains or sample spacings. It does plotting and linear algebra stuff that only happens in 2-dimensions. That said, it is supposed to look very much like an array of MeasuredFunctions, although it is missing slicing in getitem.

In MeasuredFunction, that first dimension is always going to have 2 elements – why not name them abscissa and ordinate. Operations such as crop or center of mass, act on both absc/ordi arrays, so in the example above, an operation on arr[1,_0_,:] must also have a side effect on arr[1,_1_,:]

Side note: The distinction between FunctionBundle and FunctionalBasis was always thin. Over time child functionality migrated to the parent class. Maybe they should merge into one class.

**Comment by atait** [comment]
Ok I implemented slicing.

File: lightlab/util/data/two_dim.py [hunk 1]
```diff
@@ -79,36 +95,106 @@ def __eq__(self, other):
...
+            Note:
+                Be careful about "overloading" a MeasuredFunction method in FunctionBundle.
+                It can have a different meaning than what would happen if not overloaded.
+                For example ``MeasuredFunction.__mul__(nonscalar)`` is function multiplication,
+                while ``FunctionBundle.__mul__(nonscalar)`` means vector dot product.
+
+            Todo:
+                This only works with :class:`~lightlab.util.data.one_dim.SignalProcessingMixin`
+                methods that are MF-in/MF-out. How should we handle characteristic-type methods,
+                such as ``centerOfMass``, or ``getRange``?
```
Comment (thomaslima) [inline lightlab/util/data/two_dim.py]: I'll leave this TODO ignored for now. I think we should just return a list with the results.

**Comment by thomaslima** [comment]
The tests will pass once #6 is merged. Standing by.

**Pull request merged**